### PR TITLE
add all possible ips to the serving certs

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/2075.

@liggitt

### CODE REVIEW

**Comment by sdodson** [comment]
Reading the code, this would only happen during master write-config operations, correct? Just looking to verify we're not adding this to `admin create-node-config` as those certs would most certainly be relocated to other hosts.

**Comment by deads2k** [comment]
> we're not adding this to admin create-node-config as those certs would most certainly be relocated to other hosts.

Correct.  That hostname list is built based on the nodename only or based on the explicit argument passed to create-node-config

File: pkg/cmd/server/start/master_args.go [hunk 1]
```diff
@@ -416,6 +416,17 @@ func (args MasterArgs) GetServerCertHostnames() (util.StringSet, error) {
 	}
 
 	allHostnames := util.NewStringSet("localhost", "127.0.0.1", "openshift.default.local", "kubernetes.default.local", "kubernetes-ro.default.local", masterAddr.Host, masterPublicAddr.Host, assetPublicAddr.Host)
+
+	// add the IPs that might be used based on the ListenAddr.
+	if args.ListenArg.ListenAddr.Host == "0.0.0.0" {
```
Comment (liggitt) [inline pkg/cmd/server/start/master_args.go]: do `ip := net.ParseIP(ipString)` and check `ip != nil && ip.IsUnspecified()` rather than "0.0.0.0"

**Comment by liggitt** [comment]
other than that, LGETM

**Comment by deads2k** [comment]
[merge]

**Pull request merged**